### PR TITLE
Mizar-Arktos kube-up for scaleup architecture

### PR DESCRIPTION
**Pull request by vinaykul**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**: This change updates Ubuntu 18.04 kernel to 5.6.0-rc2 that Mizar requires. It also disables predictable interface naming (Upstream mizar works with it but there is a bug in Mizar artkos integration that requires eth0 naming)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
I tested this by deploying kube-up for scale-up and scaleout with default network provider as well as with mizar as network provider.
```
root@fw0000358:~/go/src/k8s.io/varktos# export SCALEOUT_CLUSTER=false NUM_NODES=2 RUN_PREFIX=vinay-ark
root@fw0000358:~/go/src/k8s.io/varktos# export MASTER_DISK_SIZE=500GB MASTER_ROOT_DISK_SIZE=500GB KUBE_GCE_ZONE=us-west2-b MASTER_SIZE=n1-highmem-32 NODE_SIZE=n1-highmem-16 NODE_DISK_SIZE=500GB GOPATH=$HOME/go KUBE_GCE_ENABLE_IP_ALIASES=true KUBE_GCE_PRIVATE_CLUSTER=true CREATE_CUSTOM_NETWORK=true KUBE_GCE_INSTANCE_PREFIX=${RUN_PREFIX} KUBE_GCE_NETWORK=${RUN_PREFIX} ENABLE_KCM_LEADER_ELECT=false ENABLE_SCHEDULER_LEADER_ELECT=false ETCD_QUOTA_BACKEND_BYTES=8589934592 SHARE_PARTITIONSERVER=false LOGROTATE_FILES_MAX_COUNT=200 LOGROTATE_MAX_SIZE=200M KUBE_ENABLE_APISERVER_INSECURE_PORT=true KUBE_ENABLE_PROMETHEUS_DEBUG=true KUBE_ENABLE_PPROF_DEBUG=true TEST_CLUSTER_LOG_LEVEL=--v=2 HOLLOW_KUBELET_TEST_LOG_LEVEL=--v=2 GCE_REGION=us-west2-b
NETWORK_PROVIDER=mizar ./cluster/kube-up.sh

```

**Does this PR introduce a user-facing change?**: No

```release-note

```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -132,6 +135,9 @@ function main() {
 
     start-kube-apiserver
     start-kube-controller-manager
+    if [[ "${NETWORK_PROVIDER:-}" == "mizar" ]]; then
```
Comment (sonyafenge) [inline cluster/gce/gci/configure-helper.sh:138]: If this only worked for scale-up, Please add  "${SCALEOUT_CLUSTER:-false}" == "false"

**Comment by Sindica** [comment]
Please provide env set up variables and minimal test steps.

File: cluster/gce/config-default.sh [hunk 2]
```diff
@@ -96,6 +96,7 @@ NETWORK_POLICY_PROVIDER="${NETWORK_POLICY_PROVIDER:-flannel}" # calico kubenet d
 if [[ "${NETWORK_PROVIDER:-}" == "mizar" ]]; then
   NETWORK_PROVIDER_VERSION="dev"
   NETWORK_POLICY_PROVIDER="mizar"
+  CNI_BIN_DIR="/opt/cni/bin"
```
Comment (Sindica) [inline cluster/gce/config-default.sh:99]: Please add the following statement in the if block for test to pass in kube-up:
KUBE_CONTAINER_RUNTIME="docker"

Comment (vinaykul) [reply cluster/gce/config-default.sh:99]: added.

File: cluster/gce/gci/configure.sh [hunk 3]
```diff
@@ -381,8 +381,12 @@ function install-cni-network {
...
   case "${NETWORK_POLICY_PROVIDER:-flannel}" in
     mizar)
-    setup-mizar-cni-conf
-    install-mizar-cni-bin
+    if [[ "${SCALEOUT_CLUSTER:-false}" == "true" ]]; then
+      setup-mizar-cni-conf
+      install-mizar-cni-bin
+    else
+      download-mizar-cni-yaml
+    fi
```
[the remark this answers is not in the document]
Comment (vinaykul) [reply cluster/gce/gci/configure.sh:389]: see email

**Comment by vinaykul** [comment]
@sonyafenge @Sindica I have tested scaleup with default (flannel) by running kube-up.sh without any NETWORK_PROVIDER value specified and got the following results.

- With my changes in this PR:
```
root@fw0000358:~/go/src/k8s.io/varktos_jan14_rebased# ./cluster/kubectl.sh get no -owide
NAME                              STATUS                     ROLES    AGE    VERSION   INTERNAL-IP   EXTERNAL-IP    OS-IMAGE                             KERNEL-VERSION   CONTAINER-RUNTIME
vinay-arktest-master              Ready,SchedulingDisabled   <none>   3m2s   v0.9.0    10.40.0.2     34.94.108.91   Container-Optimized OS from Google   5.10.68+         containerd://1.5.4
vinay-arktest-minion-group-3trx   Ready                      <none>   3m     v0.9.0    10.40.0.3                    Container-Optimized OS from Google   5.10.68+         containerd://1.5.4
vinay-arktest-minion-group-93sp   Ready                      <none>   3m4s   v0.9.0    10.40.0.4                    Container-Optimized OS from Google   5.10.68+         containerd://1.5.4
root@fw0000358:~/go/src/k8s.io/varktos_jan14_rebased# ./cluster/kubectl.sh get po --all-namespaces -owide
NAMESPACE     NAME                                           HASHKEY               READY   STATUS              RESTARTS   AGE     IP          NODE                              NOMINATED NODE   READINESS GATES
kube-system   coredns-75c65c444f-drxqq                       3518022575354732658   1/1     Running             0          3m8s    10.64.2.9   vinay-arktest-minion-group-93sp   <none>           <none>
kube-system   etcd-empty-dir-cleanup-vinay-arktest-master    5209665295957676882   1/1     Running             0          2m29s   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   etcd-server-events-vinay-arktest-master        3723501173524514465   1/1     Running             0          2m14s   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   etcd-server-vinay-arktest-master               6497106790285606047   1/1     Running             0          2m30s   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   event-exporter-v0.2.5-868dff6494-7lggb         8262724516244272524   1/1     Running             0          3m8s    10.64.2.8   vinay-arktest-minion-group-93sp   <none>           <none>
kube-system   fluentd-gcp-scaler-74b46b8776-jz5kt            7076833373720873713   1/1     Running             0          3m7s    10.64.2.6   vinay-arktest-minion-group-93sp   <none>           <none>
kube-system   fluentd-gcp-v3.2.0-72z85                       8669113431931773301   1/1     Running             0          112s    10.40.0.3   vinay-arktest-minion-group-3trx   <none>           <none>
kube-system   fluentd-gcp-v3.2.0-dp2m7                       222142908648164623    1/1     Running             0          2m9s    10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   fluentd-gcp-v3.2.0-l5qgt                       8089845591525662348   1/1     Running             0          2m26s   10.40.0.4   vinay-arktest-minion-group-93sp   <none>           <none>
kube-system   heapster-v1.6.0-beta.1-5b955dd784-jnm27        7970294301592007111   2/2     Running             0          2m30s   10.64.1.3   vinay-arktest-minion-group-3trx   <none>           <none>
kube-system   kube-addon-manager-vinay-arktest-master        5014754618061431440   1/1     Running             0          118s    10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kube-apiserver-vinay-arktest-master            9141759218507421569   1/1     Running             0          2m38s   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kube-controller-manager-vinay-arktest-master   8130943061155147006   1/1     Running             0          2m24s   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kube-dns-autoscaler-748b78969c-j8c82           157914745518465835    1/1     Running             0          3m7s    10.64.2.5   vinay-arktest-minion-group-93sp   <none>           <none>
kube-system   kube-flannel-ds-lt8l5                          317470948335977303    0/1     Error               3          3m4s    10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kube-flannel-ds-q5p9v                          7001444317600837504   1/1     Running             0          3m6s    10.40.0.4   vinay-arktest-minion-group-93sp   <none>           <none>
kube-system   kube-flannel-ds-xsm6m                          7868157868182184570   1/1     Running             0          3m2s    10.40.0.3   vinay-arktest-minion-group-3trx   <none>           <none>
kube-system   kube-proxy-vinay-arktest-minion-group-3trx     9020506631170760401   1/1     Running             0          2m55s   10.40.0.3   vinay-arktest-minion-group-3trx   <none>           <none>
kube-system   kube-proxy-vinay-arktest-minion-group-93sp     9020506631170760401   1/1     Running             0          2m58s   10.40.0.4   vinay-arktest-minion-group-93sp   <none>           <none>
kube-system   kube-scheduler-vinay-arktest-master            7673888266147270859   1/1     Running             1          2m51s   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kubernetes-dashboard-848965699-8r2c5           8338710231154412372   1/1     Running             0          3m7s    10.64.2.2   vinay-arktest-minion-group-93sp   <none>           <none>
kube-system   l7-default-backend-6497bc5bf6-x4bpm            489009528048177722    0/1     ContainerCreating   0          3m8s    <none>      vinay-arktest-minion-group-93sp   <none>           <none>
kube-system   l7-lb-controller-v1.2.3-vinay-arktest-master   5013348416365870850   1/1     Running             0          115s    10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   metrics-server-v0.3.3-8654645975-xlhp2         5189948938107783039   2/2     Running             0          2m33s   10.64.1.2   vinay-arktest-minion-group-3trx   <none>           <none>
root@fw0000358:~/go/src/k8s.io/varktos_jan14_rebased#
```

- Without my changes against latest commit in POC branch:
```
root@fw0000358:~/go/src/k8s.io/arktos# ./cluster/kubectl.sh get no -owide
NAME                              STATUS                     ROLES    AGE    VERSION   INTERNAL-IP   EXTERNAL-IP    OS-IMAGE                             KERNEL-VERSION   CONTAINER-RUNTIME
vinay-arktest-master              Ready,SchedulingDisabled   <none>   110m   v0.9.0    10.40.0.2     34.94.88.223   Container-Optimized OS from Google   5.10.68+         containerd://1.5.4
vinay-arktest-minion-group-47cd   Ready                      <none>   110m   v0.9.0    10.40.0.4                    Container-Optimized OS from Google   5.10.68+         containerd://1.5.4
vinay-arktest-minion-group-j2hz   Ready                      <none>   110m   v0.9.0    10.40.0.3                    Container-Optimized OS from Google   5.10.68+         containerd://1.5.4
root@fw0000358:~/go/src/k8s.io/arktos# ./cluster/kubectl.sh get po --all-namespaces -owide
NAMESPACE     NAME                                           HASHKEY               READY   STATUS              RESTARTS   AGE    IP          NODE                              NOMINATED NODE   READINESS GATES
kube-system   coredns-75c65c444f-znjvz                       2529174304663452414   1/1     Running             0          110m   10.64.2.4   vinay-arktest-minion-group-47cd   <none>           <none>
kube-system   etcd-empty-dir-cleanup-vinay-arktest-master    5209665295957676882   1/1     Running             0          110m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   etcd-server-events-vinay-arktest-master        3723501173524514465   1/1     Running             0          110m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   etcd-server-vinay-arktest-master               6497106790285606047   1/1     Running             0          109m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   event-exporter-v0.2.5-868dff6494-m5twg         7873341252543253743   1/1     Running             0          110m   10.64.2.7   vinay-arktest-minion-group-47cd   <none>           <none>
kube-system   fluentd-gcp-scaler-74b46b8776-4xskr            8375669431750626738   1/1     Running             0          110m   10.64.2.3   vinay-arktest-minion-group-47cd   <none>           <none>
kube-system   fluentd-gcp-v3.2.0-cc5kv                       6439185604826925424   1/1     Running             0          109m   10.40.0.4   vinay-arktest-minion-group-47cd   <none>           <none>
kube-system   fluentd-gcp-v3.2.0-dr2g2                       3994855385708525380   1/1     Running             0          110m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   fluentd-gcp-v3.2.0-nz2pj                       931207417945828164    1/1     Running             0          110m   10.40.0.3   vinay-arktest-minion-group-j2hz   <none>           <none>
kube-system   heapster-v1.6.0-beta.1-5b955dd784-z8ndt        539324694672479471    2/2     Running             0          110m   10.64.1.3   vinay-arktest-minion-group-j2hz   <none>           <none>
kube-system   kube-addon-manager-vinay-arktest-master        5014754618061431440   1/1     Running             0          109m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kube-apiserver-vinay-arktest-master            9141759218507421569   1/1     Running             0          110m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kube-controller-manager-vinay-arktest-master   8130943061155147006   1/1     Running             0          110m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kube-dns-autoscaler-748b78969c-z9nhl           8325537797654105254   1/1     Running             0          110m   10.64.2.9   vinay-arktest-minion-group-47cd   <none>           <none>
kube-system   kube-flannel-ds-hg7h8                          3854312890351780094   1/1     Running             0          110m   10.40.0.4   vinay-arktest-minion-group-47cd   <none>           <none>
kube-system   kube-flannel-ds-p7hn5                          631230238325131094    1/1     Running             0          110m   10.40.0.3   vinay-arktest-minion-group-j2hz   <none>           <none>
kube-system   kube-flannel-ds-z9nkw                          5084984189477282137   0/1     CrashLoopBackOff    23         110m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kube-proxy-vinay-arktest-minion-group-47cd     1423854679920163009   1/1     Running             0          110m   10.40.0.4   vinay-arktest-minion-group-47cd   <none>           <none>
kube-system   kube-proxy-vinay-arktest-minion-group-j2hz     1423854679920163009   1/1     Running             0          110m   10.40.0.3   vinay-arktest-minion-group-j2hz   <none>           <none>
kube-system   kube-scheduler-vinay-arktest-master            7673888266147270859   1/1     Running             1          109m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   kubernetes-dashboard-848965699-vr42f           1628271635924776051   1/1     Running             0          110m   10.64.2.2   vinay-arktest-minion-group-47cd   <none>           <none>
kube-system   l7-default-backend-6497bc5bf6-qz2lr            2994362106888801787   0/1     ContainerCreating   0          110m   <none>      vinay-arktest-minion-group-47cd   <none>           <none>
kube-system   l7-lb-controller-v1.2.3-vinay-arktest-master   5013348416365870850   1/1     Running             0          109m   10.40.0.2   vinay-arktest-master              <none>           <none>
kube-system   metrics-server-v0.3.3-8654645975-sszgv         1257071772277343174   2/2     Running             0          110m   10.64.1.2   vinay-arktest-minion-group-j2hz   <none>           <none>
root@fw0000358:~/go/src/k8s.io/arktos# 
```

As you can see, my PR changes does not appear to introduce any new issues in arktos.

File: cluster/gce/config-default.sh [hunk 4]
```diff
@@ -377,6 +378,7 @@ NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS="${NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS:-}"
 
 CNI_VERSION="${CNI_VERSION:-}"
 CNI_SHA1="${CNI_SHA1:-}"
+CNI_BIN_DIR="${CNI_BIN_DIR:-/home/kubernetes/bin}"
```
[the remark this answers is not in the document]
Comment (vinaykul) [reply cluster/gce/config-default.sh:383]: Please see below comment https://github.com/CentaurusInfra/arktos/pull/1241#issuecomment-1016973088

File: cluster/gce/config-test.sh [hunk 5]
```diff
@@ -392,6 +392,7 @@ NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS="${NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS:-}"
 
 CNI_VERSION="${CNI_VERSION:-}"
 CNI_SHA1="${CNI_SHA1:-}"
+CNI_BIN_DIR="${CNI_BIN_DIR:-/home/kubernetes/bin}"
```
[the remark this answers is not in the document]
Comment (vinaykul) [reply cluster/gce/config-test.sh:395]: This change was introduced in order to provide the correct CNI binary dir as installed by the upstream yaml. It defaults to /home/kubernetes/bin that was hardcoded in kube-up script so it does not introduce any functional change to default CNI deployments.

Please see comment https://github.com/CentaurusInfra/arktos/pull/1241#issuecomment-1016973088 
And as per email discussion, this PR has not introduced any new problems to the 2TP-2RP default CNI deployment. As far as I can tell, there is no issue to follow-up.

File: cluster/gce/util.sh [hunk 6]
```diff
@@ -906,7 +906,7 @@ function construct-linux-kubelet-flags {
     flags+=" --register-schedulable=false"
   fi
   if [[ -n "${NETWORK_PROVIDER:-}" || -n "${NETWORK_POLICY_PROVIDER:-}" ]]; then
-    flags+=" --cni-bin-dir=/home/kubernetes/bin"
+    flags+=" --cni-bin-dir=${CNI_BIN_DIR}"
```
[the remark this answers is not in the document]
Comment (vinaykul) [reply cluster/gce/util.sh:909]: This change was introduced in order to provide the correct CNI binary dir as installed by the upstream yaml. It defaults to /home/kubernetes/bin that was hardcoded in kube-up script so it does not introduce any functional change to default CNI deployments.

Please see comment https://github.com/CentaurusInfra/arktos/pull/1241#issuecomment-1016973088 
And as per email discussion, this PR has not introduced any new problems to the 2TP-2RP default CNI deployment. As far as I can tell, there is no issue to follow-up.

File: cluster/gce/util.sh [hunk 7]
```diff
@@ -1296,6 +1296,7 @@ NODE_PROBLEM_DETECTOR_RELEASE_PATH: $(yaml-quote ${NODE_PROBLEM_DETECTOR_RELEASE
 NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS: $(yaml-quote ${NODE_PROBLEM_DETECTOR_CUSTOM_FLAGS:-})
 CNI_VERSION: $(yaml-quote ${CNI_VERSION:-})
 CNI_SHA1: $(yaml-quote ${CNI_SHA1:-})
+CNI_BIN_DIR: $(yaml-quote ${CNI_BIN_DIR:-/home/kubernetes/bin})
```
[the remark this answers is not in the document]
Comment (vinaykul) [reply cluster/gce/util.sh:1299]: This change was introduced in order to provide the correct CNI binary dir as installed by the upstream yaml. It defaults to /home/kubernetes/bin that was hardcoded in kube-up script so it does not introduce any functional change to default CNI deployments.

Please see comment https://github.com/CentaurusInfra/arktos/pull/1241#issuecomment-1016973088 
And as per email discussion, this PR has not introduced any new problems to the 2TP-2RP default CNI deployment. As far as I can tell, there is no issue to follow-up.

**Review by Sindica — APPROVED** [review]
Tested default perf test cluster up for scale up and 2x2 scale out cases. Both have hollow node started in ready status and system pods running looks ok.

**Review by sonyafenge — APPROVED** [review]
/LGTM

**Pull request merged**